File: futari-koyomi/app/banned.py
BANNED_WORDS = [
    "streak",
    "Streak",
    "STREAK",
    "連勝",
    "達成 度",
    "達成度",
    "効率",
    "義務",
    "サボり",
    "お前",
    "クソ",
    "無能",
    "ふたり 度",
    "夫婦 力",
    "夫婦力",
]


def find_banned(s: str):
    for w in BANNED_WORDS:
        if w in s:
            return w
    return None


def audit_all(strings):
    hits = []
    for s in strings:
        w = find_banned(s)
        if w:
            hits.append((s, w))
    return hits

File: inu-saiten/src/banned.js
export const BANNED_WORDS = [
  "殺す",
  "死ぬ",
  "致命",
  "劇毒",
  "クソ",
  "無能",
  "飼い主 失格",
  "飼い主失格",
  "お前",
];

export function findBanned(s) {
  for (const w of BANNED_WORDS) {
    if (s.includes(w)) return w;
  }
  return null;
}

export function auditAll(strings) {
  const hits = [];
  for (const s of strings) {
    const hit = findBanned(s);
    if (hit) hits.push({ value: s, hit });
  }
  return hits;
}
